### PR TITLE
ENGDOCS-1064

### PR DESCRIPTION
**Pull request by aevesdocker**
This PR updates the Networking in Compose page. Links now point to the correct version, the version number in the code blocks are removed and the reference info is made more clear. It also aligns the page with the Style guide.

Preview:
https://deploy-preview-16443--docsdocker.netlify.app/compose/networking/

### CODE REVIEW

**Comment by aevesdocker** [comment]
A quick PR review would be appreciated @glours 🙏

File: compose/networking.md [hunk 1]
```diff
@@ -42,14 +39,6 @@ When you run `docker compose up`, the following happens:
...
     `myapp_default` under the name `db`.
 
-> **In v2.1+, overlay networks are always `attachable`**
->
-> Starting in Compose file format 2.1, overlay networks are always created as
-> `attachable`, and this is not configurable. This means that standalone
-> containers can connect to overlay networks.
->
-> In Compose file format 3.x, you can optionally set the `attachable` property
-> to `false`.
```
Comment (glours) [inline compose/networking.md:52]: This is still true with the Compose specification, by default overlay networks are created as `attachable` by default, I think we should keep a sentence about this point (without mentioning compose file version of course 😉 )

Comment (glours) [reply compose/networking.md:52]: Ok I just saw the mention of that in the `Multi-host networking` section, ignore me 😁

**Review by glours — APPROVED** [review]
LGTM, thanks @aevesdocker

File: compose/compose-file/compose-file-v3.md [hunk 2]
```diff
@@ -1397,7 +1397,7 @@ the alias, or the service name if no alias was specified.
 
 Links are not required to enable services to communicate - by default,
 any service can reach any other service at that service’s name. (See also, the
-[Links topic in Networking in Compose](../networking.md#links).)
+[Links topic in Networking in Compose](../networking.md#link-containers).)
```
Comment (dvdksn) [inline compose/compose-file/compose-file-v3.md:1400]: not important: I am not a fan of referring to sections as "topics". I think it doesn't make sense unless you're a technical writer :D

```suggestion
[Link containers section in Networking in Compose](../networking.md#link-containers).)
```

File: compose/networking.md [hunk 3]
```diff
@@ -88,13 +81,15 @@ services:
...
 
-See the [links reference](compose-file/compose-file-v2.md#links) for more information.
+See the [links reference](compose-file/index.md#links) for more information.
 
 ## Multi-host networking
 
 When deploying a Compose application on a Docker Engine with [Swarm mode enabled](../engine/swarm/index.md),
 you can make use of the built-in `overlay` driver to enable multi-host communication.
 
+Overlay networks are always created as `attachable`.  You can optionally set the [`attachable`](compose-file/index.md#attachable) property to `false`.
```
Comment (dvdksn) [inline compose/networking.md:91]: double space

```suggestion
Overlay networks are always created as `attachable`. You can optionally set the [`attachable`](compose-file/index.md#attachable) property to `false`.
```

**Review by dvdksn — COMMENTED** [review]
minor comments ;)

File: compose/compose-file/compose-file-v3.md (hunk 2 above)
Comment (aevesdocker) [reply compose/compose-file/compose-file-v3.md:1400]: You're not wrong - changing now!

**Review by dvdksn — APPROVED** [review]

**Pull request merged**